Makefile

```
@@ -48,3 +48,7 @@ run: build
 tests:
 	pytest --rootdir=${module}/.. --cov=${module} --cov-report=xml --verbose && \
 	codecov --token=${codecov_token}
+
+frontend:
+	docker build --tag frontend_mapa_burbuja ./front && \
+	docker run -d -p 8500:80 -it frontend_mapa_burbuja 
```

Por favor agrega la versión larga de las instrucciones. ¿Serán las siguientes:
```suggestion
	docker run --detache --publish 8500:80 --interactive --tty frontend_mapa_burbuja 
```
?

---

Excelente @nepito, la simplifique un poco mas.
`docker run --detach --publish 8500:80 frontend_mapa_burbuja`

---

¿Cuáles crees que será las complicaciones de poner una prueba a la función que hiciste?
Necesitaríamos de una utilería de testing, recomiendo **Jest**, la instalaríamos y haríamos una prueba, te parece bien? 
@nepito 

- [https://jestjs.io/](https://jestjs.io/)

---

aunque recomiendo dejar las pruebas para un próximo PR,
porque si va implicar trabajo hacer esos cimientos, configuraciones extras y adaptar el proyecto actual para orientarlo a pruebas.

Aunque solo vayamos a agregar una sola prueba, va a ser todo un tema para abordar.

---

> ¿Cuáles crees que será las complicaciones de poner una prueba a la función que hiciste?
> Necesitaríamos de una utilería de testing, recomiendo **Jest**, la instalaríamos y haríamos una prueba, te parece bien?
> @nepito
> 
> * https://jestjs.io/

`jest` me parece una buena opción :+1:

---

> aunque recomiendo dejar las pruebas para un próximo PR,
> porque si va implicar trabajo hacer esos cimientos, configuraciones extras y adaptar el proyecto actual para orientarlo a pruebas.
> 
> Aunque solo vayamos a agregar una sola prueba, va a ser todo un tema para abordar.

¿Qué tanto trabajo nos puede tomar? ¿Cómo cuántos :tomato: le calculas? ¿Cómo entrarían esas pruebas, en la misma receta que ya hay de pruebas (con una modificación)? ¿En una receta distinta? ¿En un `Dockerfile` disitnto (lo pregunto porque hay muchos de esos en este repo)?

---

Actualmente y a simple vista donde se ve que se ocupa mas trabajo es en el proyecto del frontend,
mínimo unos 10 tomates,
sugiero dividir las cosas por lo menos de manera abstracta donde cada uno requerirá diferentes tipos de pruebas
(Pruebas para API y en el otro Pruebas de Código),

1.  Proyecto API de datos
2.  Proyecto API de lambdas
3.  Proyecto Frontend

   -  Orientar a pruebas e ir refactorizando poco a poco

Donde cada uno tiene su propio Dockerfile y estos son orquestados desde la raíz con del **docker compose**
hay un Dockerfile extra y aunque aun falta investigarlo mejor, creo que podrá desaparecer en algún momento.

---

Muchas gracias por tu respuesta @xwalls . No me estoy imaginando que hagamos todas las pruebas en este _pr_. Me estoy imaginado que podamos hacer la prueba de la única función que hiciste `setSeason` y dejar el resto para otro _pr_ o para cuando estemos tocando el código. Mi interés es por lo siguiente: [En las reglas para programado](https://github.com/IslasGECI/manual#reglas-para-programadoras-y-programadores) el segundo punto dice que deberíamos de verificar la resistencia a mutaciones. Entiendo que eso es mucho, pero una avanzadita hacia esa dirección no suena mal. Es por eso que pregunto cuántos tomates necesitaremos para ver la prueba de la función `setSeason` en corriendo en GitHub Actions.

---

Gracias a ti @nepito por tu respuesta, necesitamos tener un tomate al respecto, te parece bien el día lunes o martes?